pkgs/by-name/gi/ginkgo-hpc/package.nix

```diff
@@ -0,0 +1,95 @@
+{ lib
+, stdenv
+, callPackage
+, fetchFromGitHub
+, fetchpatch
+, gitUpdater
+, cmake
+, mpi
+, llvmPackages
+, papi
+, metis
+, hwloc
+, rocmUnfree ? false
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ginkgo-hpc";
+  version = "1.6.0";
+
+  src = fetchFromGitHub {
+    owner = "ginkgo-project";
+    repo = "ginkgo";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-uKUxlNlUe4HaPawEno9C57D5EFz63RB+Ei4HoIgyYjM=";
+  };
+
+  # https://github.com/ginkgo-project/ginkgo/issues/1428
+  patches = [
+    (fetchpatch {
+      name = "fix-hip-jacobi-transposition-compilation.patch";
+      url = "https://github.com/ginkgo-project/ginkgo/commit/094e58410c5fbdd7c74dbe548017d44f2b111667.patch";
+      hash = "sha256-JzKQW6XhqsBSQ0f53xuYzpkbbY5rrUFd0OvoqH57Oa0=";
+    })
+
+    (fetchpatch {
+      name = "avoid-changing-thread-indexing-scheme.patch";
+      url = "https://github.com/ginkgo-project/ginkgo/commit/a96bb1cb50217b5585abf01dab13d415e0b11b7f.patch";
+      hash = "sha256-abl7MK8j1tlbK1RPiREz82GKAKDGFY6QdjSZyQAY5Tw=";
+    })
+  ];
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [
+    finalAttrs.passthru.mpi
+    finalAttrs.passthru.openmp
+    papi
+    metis
+    hwloc
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "GINKGO_BUILD_TESTS" false)
+    (lib.cmakeBool "GINKGO_BUILD_EXAMPLES" false)
+    (lib.cmakeBool "GINKGO_BUILD_BENCHMARKS" false)
+    (lib.cmakeBool "GINKGO_BUILD_REFERENCE" true)
+    (lib.cmakeBool "GINKGO_BUILD_OMP" true)
+    (lib.cmakeBool "GINKGO_BUILD_MPI" true)
+    (lib.cmakeBool "GINKGO_MIXED_PRECISION" true)
+  ];
+
+  # For passthru overlays
+  postPatch = "";
+  postInstall = "";
+
+  passthru = {
+    inherit mpi;
+    inherit (llvmPackages) openmp;
+    cuda = finalAttrs.finalPackage.overrideAttrs (callPackage ./overlays/cuda.nix { });
+    hip = finalAttrs.finalPackage.overrideAttrs (callPackage ./overlays/hip.nix { inherit rocmUnfree; });
+    dpc = finalAttrs.finalPackage.overrideAttrs (callPackage ./overlays/dpc.nix { });
+    doc = finalAttrs.finalPackage.overrideAttrs (callPackage ./overlays/doc.nix { });
+    test = finalAttrs.finalPackage.overrideAttrs (callPackage ./overlays/test.nix { });
+    example = finalAttrs.finalPackage.overrideAttrs (callPackage ./overlays/example.nix { });
+    benchmark = finalAttrs.finalPackage.overrideAttrs (callPackage ./overlays/benchmark.nix { });
```

Comment:
the derivations in `passthru` won't be spliced because
https://github.com/NixOS/nixpkgs/issues/211340

so having one of them in `nativeBuildInputs` when cross-compiling won't pick the on build version

Reply:
So, unless I'm reading this completely wrong...
1. I need to make a scope like https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/cuda-packages.nix
2. I need to use ``makeScopeWithSplicing`` somehow?

Reply:
If those two options won't work, would just setting it to ``x86-64-linux`` for now suffice?
I don't really understand this at all.